### PR TITLE
REVIEW fix: CtAnnotation#getValue returns the default value if one exists

### LINKED ISSUE #2029: CtAnnotation.getValues() should return the default values if such exist
CtAnnotation.getValues() should return the default values if such exists

Example @Deprecated.getValues on source code returns nothing while @Deprecated.getValues on shadow classes on Java 9 boolean forRemoval, String since, see https://docs.oracle.com/javase/9/docs/api/java/lang/Deprecated.html#forRemoval--

Bug found with https://travis-ci.org/INRIA/spoon/jobs/387663415

### PR DESCRIPTION
**Pull request by monperrus**
Fix #2029

### CODE REVIEW

File: src/main/java/spoon/support/reflect/declaration/CtAnnotationImpl.java [hunk 1]
```diff
@@ -312,64 +306,92 @@ private Object getDefaultValue(String fieldName) {
...
+				return Integer.parseInt(ret.toString());
 			} else if ((type == long.class) && (ret.getClass() != long.class)) {
-				ret = Long.parseLong(ret.toString());
+				return Long.parseLong(ret.toString());
 			} else if (type == short.class && ret.getClass() != short.class) {
-				ret = Short.parseShort(ret.toString());
+				return Short.parseShort(ret.toString());
 			}
 		}
-		if (type.isArray() && ret != null && ret.getClass() != type) {
```
Comment (surli) [inline src/main/java/spoon/support/reflect/declaration/CtAnnotationImpl.java]: Are you sure with the removal of this piece of code? It seems related with the contract with annotation and arrays changed in https://github.com/INRIA/spoon/pull/1935

Comment (monperrus) [reply src/main/java/spoon/support/reflect/declaration/CtAnnotationImpl.java]: the contract of #1935 is implemented at https://github.com/INRIA/spoon/pull/2034/files#diff-94d063d96db752ab6c7a094f53c75aceR395

do you see a missing test case to specify this?

Comment (surli) [reply src/main/java/spoon/support/reflect/declaration/CtAnnotationImpl.java]: OK so you move the creation of the new array. It looks ok then. Merge?

**Comment by monperrus** [comment]
I'm done. If you don't see any problem, we can merge.

**Comment by pvojtechovsky** [comment]
Thank you ;-)

**Pull request merged**